src/components/action-bar/action-bar.types.ts

```diff
@@ -1,9 +1,16 @@
 import { MenuItem } from '../menu/menu.types';
 
-export interface ActionBarItem extends MenuItem {
-    /**
-     * Renders the button in the action bar without their labels.
-     * Does not affect the items that are overflown into the overflow menu.
-     */
-    iconOnly?: boolean;
+/**
+ * Renders the button in the action bar without their labels.
+ * Does not affect the items that are overflown into the overflow menu.
+ */
+export type ActionBarItem = ABIWithIcon | ABIWithoutIcon;
+
+interface ABIWithIcon extends MenuItem {
+    iconOnly: true;
+    icon: string;
+}
+
+interface ABIWithoutIcon extends MenuItem {
```

Comment:
I think you should name it `ActionBarItemWithLabel`. An action bar with `iconOnly: False` could still show an icon if set.